aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,531 @@
+                    return
+
+                time.sleep(retry_interval_ms)
+            except InterruptedError as e:
+                raise AwsWrapperError(Messages.get("LimitlessRouterService.InterruptedSynchronousGetRouter"), e)
+            except Exception as e:
+                if self._is_login_exception(e):
+                    raise e
+            finally:
+                retry_count += 1
+
+        raise AwsWrapperError(Messages.get("LimitlessRouterService.NoRoutersAvailable"))
+
+    def _synchronously_get_limitless_routers(self, context: LimitlessConnectionContext) -> None:
+        cache_expiration_nano: int = WrapperProperties.LIMITLESS_MONITOR_DISPOSAL_TIME_MS.get_int(context.get_props()) * 1_000_000
+
+        lock = self._force_get_limitless_routers_lock_map.compute_if_absent(
+            self._plugin_service.host_list_provider.get_cluster_id(),
+            lambda _: RLock()
+        )
+        if lock is None:
+            raise AwsWrapperError(Messages.get("LimitlessRouterService.LockFailedToAcquire"))
+
+        lock.acquire()
+        try:
+            limitless_routers = self._limitless_router_cache.get(
+                self._plugin_service.host_list_provider.get_cluster_id())
+            if limitless_routers is not None and len(limitless_routers) != 0:
+                context.set_limitless_routers(limitless_routers)
+                return
+            connection = context.get_connection()
+            if connection is None or self._plugin_service.driver_dialect.is_closed(connection):
+                context.set_connection(context.get_connect_func()())
+
+            new_limitless_routers: List[HostInfo] = self._query_helper.query_for_limitless_routers(
+                connection, context.get_host_info().port)
+
+            if new_limitless_routers is not None and len(new_limitless_routers) != 0:
+                context.set_limitless_routers(new_limitless_routers)
+                self._limitless_router_cache.compute_if_absent(
+                    self._plugin_service.host_list_provider.get_cluster_id(),
+                    lambda _: new_limitless_routers,
+                    cache_expiration_nano
+                )
+            else:
+                raise AwsWrapperError(Messages.get("LimitlessRouterService.FetchedEmptyRouterList"))
+
+        finally:
+            lock.release()
+
+    def _is_login_exception(self, error: Optional[Exception] = None):
+        self._plugin_service.is_login_exception(error)
+
+    def start_monitoring(self, host_info: HostInfo,
+                         props: Properties) -> None:
+        try:
+            limitless_router_monitor_key: str = self._plugin_service.host_list_provider.get_cluster_id()
+            cache_expiration_nano: int = WrapperProperties.LIMITLESS_MONITOR_DISPOSAL_TIME_MS.get_int(props) * 1_000_000
+            intervals_ms: int = WrapperProperties.LIMITLESS_INTERVAL_MILLIS.get_int(props)
+
+            self._limitless_router_monitor.compute_if_absent(
```

Comment:
_limitless_router_monitor should be a static variable

Reply:
Need to use `LimitlessRouterService._limitless_router_monitor` instead of `self._...`

Reply:
Same comment for _limitless_router_cache